client/src/components/workspace-nav/user-menu/UserMenu.jsx

```diff
@@ -9,6 +9,7 @@ class UserMenu extends React.Component {
   constructor(props) {
     super(props);
     this.state = { isExpanded: false };
+    this.buttonRef = React.createRef();
```

Comment:
Didn't see that `React.createRef` call before, could you expand the reasons to use it? I'm really curious

Reply:
It's because of the interaction between clicks on the button and "outside" the menu. We want to add an event listener for clicks outside of the menu and have it close. But since the button lives outside the menu it becomes "double" events being triggered. Hence we check to see if the click was outside of menu or the menu open button. https://github.com/akvo/akvo-lumen/pull/2290/files#diff-7f3fd3cbd19adb68c55c21e483c6b892R24.

Reply:
it sounds like the same use case that we have in [other places](https://github.com/akvo/akvo-lumen/search?q=ClickAway&unscoped_q=ClickAway)...  are you familiar with current component [ClickAway](https://github.com/akvo/akvo-lumen/blob/e54fccdfc12f86225c2581cf11b13488f938f02b/client/src/components/common/ClickAway.jsx) ?

Reply:
Had not seen the ClickAway but it looks like half of the issue with just clicking "somewhere" outside of the current DOM branch. I tried to cover current DOM branch and also a sister branch (the button).

Reply:
ok!